pmd-core/src/main/java/net/sourceforge/pmd/RuleSetWriter.java

```diff
@@ -266,21 +265,11 @@ private Element createPropertiesElement(List<PropertyDescriptor<?>> propertyDesc
         Element propertiesElement = null;
         if (propertyDescriptors != null) {
 
-            for (PropertyDescriptor<?> propertyDescriptor : propertyDescriptors) { // For
-                // each
-                // provided
-                // PropertyDescriptor
-
-                if (propertyDescriptor instanceof PropertyDescriptorWrapper) { // Any
-                    // wrapper
-                    // property
-                    // needs
-                    // to
-                    // go
-                    // out
-                    // as
-                    // a
-                    // definition.
+            for (PropertyDescriptor<?> propertyDescriptor : propertyDescriptors) {
+                // For each provided PropertyDescriptor
+
+                if (propertyDescriptor instanceof PropertyDescriptorWrapper) {
```

Comment:
Hmm I see, thanks for this info! I didn't really think about why wrappers were needed at all, but dropping them would really simplify the logic here. The thing is we need to make sure descriptors stay immutable. I think we'd have to make a package private constructor taking a source for each property, and initialise this source field to EXTERNAL when built with a factory, and defaulting to INTERNAL. A boolean might be enough btw